lib/https.js

```diff
@@ -34,6 +34,12 @@ const { urlToOptions, searchParamsSymbol } = require('internal/url');
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
+  }
+  opts = opts ? util._extend({}, opts) : {};
```

Comment:
If `util._extend` uses more than `Object.assign`, I think I can change it back to `util._extend`.